firefox-ios/Client/Frontend/TabContentsScripts/FormAutofillHelper/FormAutofillHelper.swift

```diff
@@ -46,24 +62,57 @@ class FormAutofillHelper: TabContentScript {
 
     // MARK: - Retrieval
 
+    /// Called when the user content controller receives a script message.
+    ///
+    /// - Parameters:
+    ///   - userContentController: The user content controller.
+    ///   - message: The script message received.
     func userContentController(_ userContentController: WKUserContentController,
                                didReceiveScriptMessage message: WKScriptMessage) {
         // Note: We require frame so that we can submit information
         // to an embedded iframe on a webpage for injecting card info
         frame = message.frameInfo
 
-        guard let data = getValidPayloadData(from: message),
-              let fieldValues = parseFieldType(messageBody: data),
-              let payloadType = FormAutofillPayloadType(rawValue: fieldValues.type)
-        else {
-            logger.log("Unable to find the payloadType for the credit card JS input",
-                       level: .warning,
-                       category: .webview)
-            return
-        }
+        // Check if tabWebView is available
+        guard let tabWebView = tab?.webView else { return }
+
+        switch HandlerName(rawValue: message.name) {
+        case .addressFormMessageHandler:
```

Comment:
Since now the mechanism for handling the message has been updated, it would be good to update `FormAutofillHelperTests.swift`

Reply:
tests are passing